### PR TITLE
[Merged by Bors] - mirrord console

### PR DESCRIPTION
**Pull request by aviramha**

### CODE REVIEW

File: mirrord/cli/src/extension.rs [hunk 1]
```diff
@@ -6,7 +6,6 @@ use mirrord_progress::{Progress, ProgressMode, TaskProgress};
 use crate::{config::ExtensionExecArgs, error::CliError, execution::MirrordExecution, Result};
 
 /// Faciliate the execution of a process using mirrord by an IDE extension
```
Comment (eyalb181) [inline mirrord/cli/src/extension.rs:8]:
```suggestion
/// Facilitate the execution of a process using mirrord by an IDE extension
```

File: mirrord/cli/src/extension.rs [hunk 2]
```diff
@@ -6,7 +6,6 @@ use mirrord_progress::{Progress, ProgressMode, TaskProgress};
...
-#[tokio::main(flavor = "current_thread")]
```
Comment (eyalb181) [inline mirrord/cli/src/extension.rs:9]: Why was this necessary?

File: mirrord/console/src/main.rs [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+    extract::ws::{Message, WebSocket, WebSocketUpgrade},
+    response::Response,
+    routing::get,
+    Router,
+};
+use protocol::Hello;
+use tracing::{error, info, metadata::LevelFilter};
+use tracing_subscriber::{fmt::format::FmtSpan, prelude::*};
+
+struct WsWrapper {
```
Comment (eyalb181) [inline mirrord/console/src/main.rs:15]:
```suggestion
struct WebSocketWrapper {
```
Looks like the convention is to spell it out

File: mirrord/console/src/main.rs [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+    let mut wrapper = WsWrapper::new(socket);
+
+    let client_info: Hello = match wrapper.next_message().await {
+        Some(hello) => {
+            info!("Client connected -  process info {hello:?}");
+            hello
+        }
+        None => {
+            error!("Client disconnected");
+            return;
```
Comment (eyalb181) [inline mirrord/console/src/main.rs:62]: Is there a reconnect mechanism? If there isn't, don't we want to crash here?

File: mirrord/console/src/error.rs [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+pub type Result<T, E = ConsoleError> = std::result::Result<T, E>;
+
+#[derive(Error, Diagnostic, Debug)]
+pub enum ConsoleError {
+    #[error("failed to connect to console `{0}`")]
+    #[diagnostic(help("Please check that the console is running and address is correct."))]
+    ConnectError(#[source] tokio_tungstenite::tungstenite::Error),
+    #[error("WS Socket occured after connection `{0}`")]
+    #[diagnostic(help("Please report a bug."))]
+    WsSocketError(#[from] tokio_tungstenite::tungstenite::Error),
```
Comment (eyalb181) [inline mirrord/console/src/error.rs:13]:
```suggestion
    WebSocketError(#[from] tokio_tungstenite::tungstenite::Error),
```

File: mirrord/cli/src/extension.rs (hunk 2 above)
Comment (aviramha) [reply mirrord/cli/src/extension.rs:9]: We didn't have an event loop in the caller, and now we have.

File: mirrord/console/src/main.rs (hunk 4 above)
Comment (aviramha) [reply mirrord/console/src/main.rs:62]: it's logging, I don't think a crash is necessary.

File: mirrord/console/src/main.rs [hunk 6]
```diff
@@ -0,0 +1,111 @@
...
+};
+use protocol::Hello;
+use tracing::{error, info, metadata::LevelFilter};
+use tracing_subscriber::{fmt::format::FmtSpan, prelude::*};
+
+struct WebSocketWrapper {
+    socket: WebSocket,
+}
+
+impl WsWrapper {
```
Comment (eyalb181) [inline mirrord/console/src/main.rs:19]:
```suggestion
impl WebSocketWrapper {
```

**Review by eyalb181 — CHANGES_REQUESTED** [review]

**Review by eyalb181 — APPROVED** [review]
